ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ParquetHiveSerDe.java

```diff
@@ -91,11 +93,61 @@ public class ParquetHiveSerDe extends AbstractSerDe implements SchemaInference {
 
   private ObjectInspector objInspector;
   private ParquetHiveRecord parquetRow;
+  private ObjectInspectorConverters.Converter converter;
 
   public ParquetHiveSerDe() {
     parquetRow = new ParquetHiveRecord();
   }
 
+  // Recursively check if CHAR or VARCHAR types are used
+  private boolean needsConversion(TypeInfo type) {
```

Comment:
Thinking about the number of times needsConversion() and the subsequent convert() would be called for a table scan that is reading say N parquet files each with m row groups.  Such operations on a per file or per row group would be ok but if done on per row basis, it would be a perf hit.

Reply:
Yes - I agree it could be a perf hit here. Though it wouldn't surprise me if the perf difference is small since these paths already do many allocations per row.

BUT - I did change my approach with what I think is a much more appropriate solution that hopefully avoids the extra allocation.

Reply:
I like the revised approach which does not do the extra allocation.